src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -500,6 +501,8 @@ private static BaseListSyntax GetBaseListOpt(SingleTypeDeclaration decl)
                     baseType = baseBinder.BindType(typeSyntax, diagnostics, newBasesBeingResolved).Type;
                 }
 
+                checkPrimaryConstructorBaseType(baseTypeSyntax, baseType);
```

Comment:
>checkPrimaryConstructorBaseType [](http://example.com/codeflow?start=16&length=31)

It is probably not worth running this check past the first item, or we should complain unconditionally (only first item in the list is allowed to have arguments).

Reply:
The parser only produces a PrimaryConstructorBaseType for the first element, so this check bails out on subsequent items already. I'm not sure what you're proposing.

Reply:
Added a check, but seems redundant to me...

Reply:
> Added a check, but seems redundant to me...

I don't think it is redundant because we are not always deal with trees from the parser.